File: Programmers/hash/완주하지 못한 선수.js
function solution(participant, completion) {
    var answer = '';
    var obj = {};
// 객체로 만듬
    for(var i=0;i<completion.length;i++){
        if(!obj[completion[i]]) {
            obj[completion[i]] = 1;
        } else {
            obj[completion[i]]=obj[completion[i]]+1;
        }
    }

    for(var i=0;i<participant.length;i++){
        if(!obj[participant[i]] || obj[participant[i]]==0) {
            return participant[i];
        } else {
            obj[participant[i]]=obj[participant[i]]-1;
        }
    }
}

File: Programmers/hash/완주하지 못한 선수.py
def solution(participant, completion):
    d = {}
    for i in completion : 
        d[i]=d.get(i,0)+1

    for i in participant:
        if(d.get(i,0)==0):
            return i
        else:
            d[i]=d[i]-1
    return


participant = ["leo", "kiki", "eden"]
completion = ["eden", "kiki"]
solution(participant, completion)

participant = ["marina", "josipa", "nikola", "vinko", "filipa"]
completion = ["josipa", "filipa", "marina", "nikola"]
solution(participant, completion)

participant = ["mislav", "stanko", "mislav", "ana"]
completion = ["stanko", "ana", "mislav"]
solution(participant, completion)
